Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -483,4 +490,168 @@ theorem intValuation_X : (idealX K).intValuationDef X = ↑(Multiplicative.ofAdd
 
 end PowerSeries
 
+namespace RatFunc
+
+open IsDedekindDomain.HeightOneSpectrum PowerSeries
+
+theorem valuation_eq_LaurentSeries_valuation (P : RatFunc K) :
+    (Polynomial.idealX K).valuation P = (PowerSeries.idealX K).valuation (P : LaurentSeries K) := by
+  refine' RatFunc.induction_on' P _
+  intro f g h
+  rw [Polynomial.valuation_of_mk K f h, RatFunc.mk_eq_mk' f h, Eq.comm]
+  convert @valuation_of_mk' (PowerSeries K) _ _ (LaurentSeries K) _ _ _ (PowerSeries.idealX K) f
+        ⟨g, mem_nonZeroDivisors_iff_ne_zero.2 <| coe_ne_zero h⟩
+  · simp only [IsFractionRing.mk'_eq_div, coe_div, LaurentSeries.coe_algebraMap, coe_coe]
+    rfl
+  exacts [intValuation_eq_of_coe _, intValuation_eq_of_coe _]
+
+end RatFunc
+
+namespace LaurentSeries
+
+open IsDedekindDomain.HeightOneSpectrum PowerSeries RatFunc
+
+instance : Valued (LaurentSeries K) ℤₘ₀ := Valued.mk' (PowerSeries.idealX K).valuation
+
+theorem valuation_X_pow (s : ℕ) :
+    Valued.v (((X : K⟦X⟧) : LaurentSeries K) ^ s) = Multiplicative.ofAdd (-(s : ℤ)) := by
+  erw [map_pow,/-  this, -/ ← one_mul (s : ℤ), ← neg_mul (1 : ℤ) s, Int.ofAdd_mul,
```

Comment:
```suggestion
  erw [map_pow, ← one_mul (s : ℤ), ← neg_mul (1 : ℤ) s, Int.ofAdd_mul,
```

Reply:
Also, do you understand where exactly you need the `e` in `erw`? Is this pointing at a missing API lemma?